targets/TARGET_Samsung/TARGET_SIDK_S5JS100/device/s5js100_systemreset.c

```diff
@@ -0,0 +1,325 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/****************************************************************************
+ *
+ * Copyright 2019 Samsung Electronics All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+/* @file  : s5js100_systemreset.c
+ * @brief : board reset source file
+ * @date  : June 2019
+ *
+ * @note  : Add chip dependent feature and more detail error status
+ *
+ */
+
+/****************************************************************************
+  Included Files
+ ****************************************************************************/
+#include "cmsis.h"
+#include "mbed_crash_data_offsets.h"
+#include "mbed_retarget.h"
+#include "mbed_critical.h"
+#include "mbed_error.h"
+#include "mbed_error_hist.h"
+#include "mbed_interface.h"
+#include "mbed_power_mgmt.h"
+#include "mbed_stats.h"
+#include "mbed_atomic.h"
+#include "mbed_fault_handler.h"
+
+#include "rtx_os.h"
+#include <inttypes.h>
+#include <string.h>
+
+#ifndef NDEBUG
+#define ERROR_REPORT(ctx, error_msg, error_filename, error_line) print_error_report(ctx, error_msg, error_filename, error_line)
+static void print_error_report(const mbed_error_ctx *ctx, const char *, const char *error_filename, int error_line);
+#else
+#define ERROR_REPORT(ctx, error_msg, error_filename, error_line) ((void) 0)
+#endif
+
+extern mbed_fault_context_t *mbed_fault_context;
+static int error_count = 0;
+static mbed_error_ctx first_error_ctx = {0};
+static mbed_error_ctx last_error_ctx = {0};
+static mbed_error_hook_t error_hook = NULL;
+static mbed_error_status_t handle_error(mbed_error_status_t error_status, unsigned int error_value, const char *filename, int line_number, void *caller);
+
+static inline const char *name_or_unnamed(const char *name)
+{
+    return name ? name : "<unnamed>";
+}
+
+
+#ifndef NDEBUG
+#define GET_TARGET_NAME_STR(tgt_name)   #tgt_name
+#define GET_TARGET_NAME(tgt_name)       GET_TARGET_NAME_STR(tgt_name)
+static void print_error_report(const mbed_error_ctx *ctx, const char *error_msg, const char *error_filename, int error_line)
```

Comment:
why is this copy from `mbed_error.c` file? Some of the error functionality is duplicated here.

Reply:
Now every mbed_error.c duplicate code is removed as shown in https://github.com/ARMmbed/mbed-os/pull/12106/files#diff-25aa845838d43d1ae1c81f7378a0ef67R102.